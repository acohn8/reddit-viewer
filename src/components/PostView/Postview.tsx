import * as React from 'react';
import { Divider, Header, Image, Modal, Segment } from 'semantic-ui-react';

import CommentContainer from '../../containers/CommentsContainer/CommentContainer';

export interface Props {
  children?: string;
  title: string;
  image: string;
  permalink: string;
  date: string;
}

const PostView = (props: Props) => {
  const {
    title, image, children, date, permalink,
  } = props;
  return (
    <Modal
      dimmer="blurring"
      size="large"
      trigger={
        <Header size="medium" as="a">
          {children}
          <Header.Subheader>{date}</Header.Subheader>
        </Header>
      }
    >
      <Modal.Header>{title}</Modal.Header>
      <Segment style={{ maxWidth: 400, margin: 'auto' }}>
        <Image size="huge" src={image} wrapped />
      </Segment>
      <Divider />
      <Modal.Content image scrolling>
        <Modal.Description>
          <Header>Comments</Header>
          <CommentContainer permalink={permalink} />
        </Modal.Description>
      </Modal.Content>
    </Modal>
  );
};

export default PostView;
