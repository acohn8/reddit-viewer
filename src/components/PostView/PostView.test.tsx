import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';

import  PostView  from './Postview';

describe('PostView', () => {
  const output = shallow(<PostView title="breaking" image="image.jpg" permalink='reddit.com' date='yesterday'  />
  it('should render correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should have a title', () => {
    expect(output.find('breaking'));
  });
  it('should have an image', () => {
    expect(output.find('image.jpg'));
  });
});
