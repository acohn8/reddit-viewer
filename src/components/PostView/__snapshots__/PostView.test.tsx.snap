// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostView should render correctly 1`] = `
<Modal
  centered={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer="blurring"
  eventPool="Modal"
  size="large"
  trigger={
    <Header
      as="a"
      onClick={[Function]}
      size="medium"
    >
      <HeaderSubheader>
        yesterday
      </HeaderSubheader>
    </Header>
  }
>
  <ModalHeader>
    breaking
  </ModalHeader>
  <Segment
    style={
      Object {
        "margin": "auto",
        "maxWidth": 400,
      }
    }
  >
    <Image
      as="img"
      size="huge"
      src="image.jpg"
      ui={true}
      wrapped={true}
    />
  </Segment>
  <Divider />
  <ModalContent
    image={true}
    scrolling={true}
  >
    <ModalDescription>
      <Header>
        Comments
      </Header>
      <Connect(CommentContainer) />
    </ModalDescription>
  </ModalContent>
</Modal>
`;
